app/pages/patientdata/patientdata.js

```diff
@@ -504,13 +485,20 @@ export let PatientData = React.createClass({
 
     var refresh = this.props.onRefresh;
     if (refresh) {
-      this.setState({title: this.DEFAULT_TITLE});
-      refresh();
+      this.props.clearPatientData(this.props.routeParams.id);
```

Comment:
Hmm, just tried and I've got the GIF looping indefinitely. Will try to read through and see what I can see.